modules/packages/UnrolledLinkedList.chpl

```diff
@@ -0,0 +1,1245 @@
+            }
+            ref dst = cur!.data[dst_i];
+            if src_i != dst_i then dst = src;
+            src_i += 1;
+            dst_i += 1;
+          }
+
+          cur!.size = dst_i;
+
+          cur = cur!.next;
+        }
+
+        // Maybe merge nodes
+        cur = _head;
+        while cur != nil {
+          try! {
+            var result = _merge(cur!);
+            if result == false then cur = cur!.next;
+          }
+        }
+
+        result = removed;
+        _size -= removed;
+      }
+
+      return result;
+    }
+
+    pragma "no doc"
+    proc ref _popAtIndex(idx: int): eltType {
+      _size -= 1;
+
+      var cur = _head;
+      var pos = 0;
+      while cur != nil {
+        if idx < pos + cur!.size {
+          var result = cur!.data[idx-pos];
+
+          // Shift elements
+          cur!.size -= 1;
+          for i in idx-pos..<cur!.size {
+            cur!.data[i] = cur!.data[i+1];
+          }
+
+        try! {
+          _merge(cur: unmanaged _linkedNode(eltType));
+        }
+
+          return result;
+        }
+        else {
+          pos += cur!.size;
+        }
+        cur = cur!.next;
+      }
+
+      // shouldn't reach here
+      boundsCheckHalt("out of bounds on unrolledLinkedList");
+    }
+
+    /*
+      Remove the element at the end of this unrolledLinkedList and return it.
+
+      .. warning::
+
+        Popping an element from this unrolledLinkedList will invalidate any reference to
+        the element taken while it was contained in this unrolledLinkedList.
+
+      .. warning::
+
+        Calling this method on an empty unrolledLinkedList will cause the currently running
+        program to halt. If the `--fast` flag is used, no safety checks will
+        be performed.
+
+      :return: The element popped.
+      :rtype: `eltType`
+    */
+    proc ref pop(): eltType {
+      _enter(); defer _leave();
+
+      if boundsChecking && _size <= 0 {
+        boundsCheckHalt("Called \"unrolledLinkedList.pop\" on an empty unrolledLinkedList.");
+      }
+
+      var result = _popAtIndex(_size-1);
+      return result;
+    }
+
+    /*
+      Remove the element at the index `idx` from this unrolledLinkedList and return it. The
+      elements at indices after `idx` are shifted one to the left in memory,
+      making this operation O(n).
+
+      .. warning::
+
+        Popping an element from this unrolledLinkedList will invalidate any reference to
+        the element taken while it was contained in this unrolledLinkedList.
+
+      .. warning::
+
+        Calling this method on an empty unrolledLinkedList or with values of `idx` that
+        are out of bounds will cause the currently running program to halt.
+        If the `--fast` flag is used, no safety checks will be performed.
+
+      :arg idx: The index of the element to remove.
+      :type idx: `int`
+
+      :return: The element popped.
+      :rtype: `eltType`
+    */
+    proc ref pop(idx: int): eltType {
```

Comment:
I don't remember if this came up before, but when I see `pop()`, I think of a stack, where `pop()` means only removing the top element, not an arbitrary one.

Reply:
This happens with `List`

Reply:
I second Paul on this. Frankly, I am surprised to see a method with the similar signature in List. Checking if there's anything to it. In any case, we can keep it for now for consistency.